### ISSUE TITLE
Support Gatsby plugins with resolve

### ISSUE THREAD

**Issue by znarf** [depcheck/depcheck#525]
In the following file, https://github.com/badges/shields/blob/master/gatsby-config.js, `gatsby-plugin-page-creator` is not detected because it's set using a different syntax than what is currently expected.

``` 
 plugins: [
    {
      resolve: 'gatsby-plugin-page-creator',
      options: {
        path: `${__dirname}/frontend/pages`,
      },
    },
    'gatsby-plugin-react-helmet',
    'gatsby-plugin-catch-links',
    'gatsby-plugin-styled-components',
    'gatsby-plugin-remove-trailing-slashes',
    'gatsby-plugin-typescript',
```

**Comment by nagygergo** [comment]
Also, this is a valid config too, so resolving the used plugins seems to be a harder task:
``` javascript
 'gatsby-plugin-sharp',
'gatsby-transformer-sharp',
{
      resolve: 'gatsby-transformer-remark',
      options: {
        plugins: [
          {
            resolve: 'gatsby-remark-relative-images',
            options: {
              name: 'uploads',
            },
          }]
     }
}

**Comment by nagygergo** [comment]
Gatsby identifies the following elements as plugin entries:
- `gatsby-plugin-sharp`
- `{ resolve: 'gatsby-transformer-sharp', options: {} }`
- `{ resolve: 'gatsby-transformer-remark' }`

Also there is an option for plugins to have plugins of their own. So the based on the above snippets I'd consider the following rules to find the used plugins:
- Plugin entry must be an element of an array that is under a `plugins` key. So `plugins: [<pluginEntry>]`.
- Plugin entry must be one of the above shapes above.
